app/uk/gov/hmrc/organisationsdetailsapi/connectors/OrganisationsMatchingConnector.scala

```diff
@@ -27,17 +28,25 @@ import java.util.UUID
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-class OrganisationsMatchingConnector @Inject() (httpClient: HttpClient, servicesConfig: ServicesConfig) {
+class OrganisationsMatchingConnector @Inject() (httpClient: HttpClientV2, servicesConfig: ServicesConfig) {
 
   private val serviceUrl = servicesConfig.baseUrl("organisations-matching-api")
 
-  def resolve(matchId: UUID)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[OrganisationMatch] =
-    httpClient.GET[OrganisationMatch](s"$serviceUrl/match-record/$matchId") recover {
-      case UpstreamErrorResponse(_, 404, _, _) => throw new MatchNotFoundException
+  def resolve(matchId: UUID)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[OrganisationMatch] = {
+    httpClient
+      .get(url"$serviceUrl/match-record/$matchId")
+      .execute[OrganisationMatch]
+      .recover {
+        case UpstreamErrorResponse(_, 404, _, _) => throw new MatchNotFoundException
```

Comment:
This can refactored as we are using same `case` and `throw` in handling `resolveVat` method below